python/ribasim/ribasim/geometry/edge.py

```diff
@@ -112,13 +123,46 @@ def add(
             index=pd.Index([edge_id], name="edge_id"),
         )
 
+        self._validate_edge(to_node, from_node, edge_type)
+
         self.df = GeoDataFrame[EdgeSchema](pd.concat([self.df, table_to_append]))
         if self.df.duplicated(subset=["from_node_id", "to_node_id"]).any():
             raise ValueError(
                 f"Edges have to be unique, but edge with from_node_id {from_node.node_id} to_node_id {to_node.node_id} already exists."
             )
         self._used_edge_ids.add(edge_id)
 
+    def _validate_edge(self, to_node: NodeData, from_node: NodeData, edge_type: str):
+        assert self.df is not None
+        in_neighbor: int = self.df.loc[
+            (self.df["to_node_id"] == to_node.node_id)
+            & (self.df["edge_type"] == edge_type)
+        ].shape[0]
+
+        out_neighbor: int = self.df.loc[
+            (self.df["from_node_id"] == from_node.node_id)
+            & (self.df["edge_type"] == edge_type)
+        ].shape[0]
+        # validation on neighbor amount
+        if edge_type == "flow":
+            if in_neighbor >= flow_edge_neighbor_amount[to_node.node_type][1]:
```

Comment:
In all these lines below I see quite some code duplication. Would be good to give `flow_edge_neighbor_amount[to_node.node_type][1]` a variable and reuse it on the line below. And do that for every of these checks.